src/components/SuperLikeContentCard.vue

```diff
@@ -0,0 +1,247 @@
+
+export default {
+  name: 'SuperLikeContentCard',
+  components: {
+    BookmarkFilledIcon,
+    BookmarkOutlinedIcon,
+    Button,
+    Card,
+    Identity,
+    Placeholder,
+  },
+  props: {
+    preset: {
+      type: String,
+      default: 'default',
+    },
+    referrer: {
+      type: String,
+      required: true,
+    },
+    authorId: {
+      type: String,
+      default: undefined,
+    },
+    title: {
+      type: String,
+      default: undefined,
+    },
+    description: {
+      type: String,
+      default: undefined,
+    },
+    coverSrc: {
+      type: String,
+      default: undefined,
+    },
+    likeCount: {
+      type: Number,
+      default: -1,
+    },
+    superLikeId: {
+      type: String,
+      default: undefined,
+    },
+    superLikeShortId: {
+      type: String,
+      default: undefined,
+    },
+    timestamp: {
+      type: Number,
+      default: 0,
+    },
+  },
+  data() {
+    return {
+      isLoading: true,
+
+      author: { user: this.authorId },
+      internalUrl: this.referrer,
+      internalTitle: this.title,
+      internalDescription: this.description,
+      internalCoverSrc: this.coverSrc,
+      internalLikeCount: this.likeCount,
+    };
+  },
+  computed: {
+    ...mapGetters([
+      'getUserId',
+      'getUserInfoById',
+      'getArticleInfoByReferrer',
+      'getIsInBookmark',
+      'getIsFollowedAuthor',
+    ]),
+
+    shouldFetchArticle() {
+      return !this.internalTitle;
+    },
+    href() {
+      if (this.superLikeShortId) {
+        return getSuperLikeRedirectLink(this.superLikeShortId);
+      }
+      if (this.superLikeId) return getSuperLikeRedirectLink(this.superLikeId);
+      return this.internalUrl;
+    },
+  },
+
+  methods: {
+    ...mapActions([
+      'fetchUserInfo',
+      'fetchArticleInfo',
+      'addBookmark',
+      'removeBookmark',
+      'followAuthor',
+      'unfollowAuthor',
+    ]),
+
+    formatDate(timestamp) {
+      return dateFormat(new Date(timestamp), 'DD/MM/YYYY');
+    },
+
+    async fetchContent() {
+      try {
+        this.isLoading = true;
+        const promises = [];
+        if (
+          this.shouldFetchArticle &&
+          !this.getArticleInfoByReferrer(this.referrer)
+        ) {
+          promises.push(
+            this.fetchArticleInfo(this.referrer)
+              .then(() => this.updateArticleInfo())
+              .catch(() => ({}))
+          );
+        } else {
+          this.updateArticleInfo();
+        }
+        if (!this.author.user) await Promise.all(promises);
+        if (this.author.user && !this.getUserInfoById(this.author.user)) {
+          promises.push(
+            this.fetchUserInfo(this.author.user)
+              .then(() => this.updateAuthorInfo())
+              .catch(() => ({}))
+          );
+        } else {
+          this.updateAuthorInfo();
+        }
+        await Promise.all(promises);
+      } finally {
+        this.$emit('fetched');
```

Comment:
Should the component still emit `fetched` if any of the promise throws?

Reply:
This is for parent component to recalculate view size after loading